src/PKSim.Core/Snapshots/Mappers/QualificationStepMapper.cs

```diff
@@ -0,0 +1,91 @@
+﻿using System;
+using System.Threading.Tasks;
+using OSPSuite.Core.Domain;
+using OSPSuite.Core.Services;
+using PKSim.Assets;
+using PKSim.Core.Model;
+using SnapshotQualificationStep = PKSim.Core.Snapshots.QualificationStep;
+
+namespace PKSim.Core.Snapshots.Mappers
+{
+   public class QualificationStepMapper : SnapshotMapperBase<IQualificationStep, SnapshotQualificationStep, PKSimProject>
+   {
+      private readonly ILogger _logger;
+      private const string QUALIFICATION_STEP_SUFFIX = "QualificationStep";
+
+      public QualificationStepMapper(ILogger logger)
+      {
+         _logger = logger;
+      }
+
+      public override async Task<SnapshotQualificationStep> MapToSnapshot(IQualificationStep qualificationStep)
+      {
+         var snapshot = await SnapshotFrom(qualificationStep, x => { x.Type = friendlyNameFor(qualificationStep.GetType().Name); });
+         mapQualificationStepPropertiesToSnapsot(snapshot, qualificationStep);
+         return snapshot;
+      }
+
+      private void mapQualificationStepPropertiesToSnapsot(SnapshotQualificationStep snapshot, IQualificationStep qualificationStep)
+      {
+         switch (qualificationStep)
+         {
+            case RunParameterIdentificationQualificationStep runParameterIdentification:
```

Comment:
This willl need to be extended for all qualification plan step we will support. Depending on number, this may or may not have to be refactored to a separate class